string_case_test.go

```diff
@@ -0,0 +1,53 @@
+package microerror
+
+import (
+	"testing"
+)
+
+func Test_toStringCase(t *testing.T) {
+	testCases := []struct {
+		Name           string
+		InputString    string
+		ExpectedString string
+	}{
+		{
+			Name:           "case 0: camel case to string case with lower start",
+			InputString:    "fooBar",
+			ExpectedString: "foo bar",
+		},
+		{
+			Name:           "case 1: camel case to string case with lower start and longer input",
+			InputString:    "fooBarBazupKick",
+			ExpectedString: "foo bar bazup kick",
+		},
+		{
+			Name:           "case 2: camel case to string case with upper start",
+			InputString:    "FooBar",
+			ExpectedString: "foo bar",
+		},
+		{
+			Name:           "case 3: camel case to string case with upper start and longer input",
+			InputString:    "FooBarBazupKick",
+			ExpectedString: "foo bar bazup kick",
+		},
+		{
+			Name:           "case 4: real private error kind",
+			InputString:    "authenticationError",
+			ExpectedString: "authentication error",
+		},
+		{
+			Name:           "case 5: real public error kind",
+			InputString:    "AuthenticationError",
+			ExpectedString: "authentication error",
+		},
```

Comment:
Would be good to also respect all uppercase abbreviations:

```
		{
			Name:           "case 6: camel case with abbreviation",
			InputString:    "invalidHTTPStatusError",
			ExpectedString: "invalid http status",
		},
```

Reply:
Extended the tests.